### PR TITLE
feat: add c# serializer and deserializer functionality

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR adds functionality to render serializer and deserializer functionality to C# class output. This will enable correct behavior from `System.Text.Json.JsonSerializer.Serialize` and `System.Text.Json.JsonSerializer.Deserialize`, so you can do the following in your code:
```c#
TestModel temp = new TestModel
{
    ...
};
string json = JsonSerializer.Serialize(temp);
TestModel output = JsonSerializer.Deserialize<TestModel>(json);
```

**Related issue(s)**
Partly solves https://github.com/asyncapi/modelina/issues/262

### CODE REVIEW

**Comment by jonaslagoni** [comment]
Blocked by #333

File: src/generators/csharp/presets/JsonSerializerPreset.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      renderer.addDependency('using System.Text.Json;');
+      renderer.addDependency('using System.Text.Json.Serialization;');
+      renderer.addDependency('using System.Text.RegularExpressions;');
+      
+      const formattedModelName = renderer.nameType(model.$id);
+      const unmarshal = renderUnmarshal({renderer, model});
+      const marshal = renderMarshal({renderer, model});
+
+      return `[JsonConverter(typeof(${formattedModelName}Converter))]
+${await renderer.defaultSelf()}
```
Comment (magicmatatjahu) [inline src/generators/csharp/presets/JsonSerializerPreset.ts:231]: You can use the content from previous preset:

```suggestion
${content}
```

but then you must add `content` in the arguments  of `self` function.

**Review by magicmatatjahu — COMMENTED** [review]
If you called functionality as serializer, then all TS functions should also be called `serialize`, not `marshal` :) or convert.

**Comment by jonaslagoni** [comment]
> If you called functionality as serializer, then all TS functions should also be called `serialize`, not `marshal` :) or convert.

Hmm, I see your point, problem is, for C# it is part of the default dependencies `System.Text.Json` meaning it is called serializer, nothing I can change about that 😄 We could change the naming for C#, however in that language many refer to it as un/marshal. Not sure it is a problem? 🤔

File: src/generators/csharp/presets/JsonSerializerPreset.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/generators/csharp/presets/JsonSerializerPreset.ts:231]: Done 👍

**Comment by magicmatatjahu** [comment]
> Hmm, I see your point, problem is, for C# it is part of the default dependencies System.Text.Json meaning it is called serializer, nothing I can change about that 😄 We could change the naming for C#, however in that language many refer to it as un/marshal. Not sure it is a problem? 🤔

We can left the `marshal` naming. It was my first concern when I saw marshal names in source code and `Serializer` in the output 😄

**Review by magicmatatjahu — APPROVED** [review]
LGTM

**Comment by jonaslagoni** [comment]
> > Hmm, I see your point, problem is, for C# it is part of the default dependencies System.Text.Json meaning it is called serializer, nothing I can change about that 😄 We could change the naming for C#, however in that language many refer to it as un/marshal. Not sure it is a problem? 🤔
> 
> We can left the `marshal` naming. It was my first concern when I saw marshal names in source code and `Serializer` in the output 😄

Ahh! Let's change the naming inside the file, to make it consistent 👍 I thought you meant that the TS output has marshal and C# has sterilizer 😄

**Comment by magicmatatjahu** [comment]
@jonaslagoni Yeah, exactly

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**